### ISSUE TITLE
Add .js suffix to "main", "module" and "jsnext:main" paths in package.json

### ISSUE THREAD

**Issue by Lebed-kun** [recharts/recharts#2858]
I'm proposing it for wider compatibility with existing configurations. Without it, some Webpack configs generate following error:

```
Error: Cannot find module '<some-path-to-project>/node_modules/recharts/lib/index'. Please verify that the package.json has a valid "main" entry
```

Most other libs have ".js" suffix on their entrypoints

**Comment by ckifer** [comment]
this will be fixed in 3.x unless https://github.com/recharts/recharts/pull/4429 is backported, thanks!

**Comment by ckifer** [comment]
cool, now will be fixed next patch

**Comment by ckifer** [comment]
fixed in 2.12.6
